native/packages/react-native-bpk-component-image/src/BpkImage.js

```diff
@@ -0,0 +1,178 @@
+    width: '100%',
+    height: '100%',
+    backgroundColor: colorGray300,
+    justifyContent: 'center',
+    alignItems: 'center',
+    position: 'absolute',
+  },
+});
+
+export type Props = {
+  onLoad: ?() => mixed,
+  style: ?StyleObj,
+  alt: string,
+  borderRadius: boolean,
+  inView: boolean,
+  loaded: boolean,
+  source: ImageSourcePropType,
+};
+
+type State = {
+  imageOpacity: number,
+  loadingIndicatorOpacity: number,
+  showLoadingIndicator: boolean,
+};
+
+class BpkImage extends Component<Props, State> {
+  static propTypes = {
+    alt: PropTypes.string.isRequired,
+    source: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
+      .isRequired,
+    borderRadius: PropTypes.bool,
+    inView: PropTypes.bool,
+    loaded: PropTypes.bool,
+    onLoad: PropTypes.func,
+    style: ViewPropTypes.style,
+  };
+
+  static defaultProps = {
+    borderRadius: true,
+    inView: true,
+    loaded: true,
+    onLoad: null,
+    style: null,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      showLoadingIndicator: !props.loaded,
+      imageOpacity: new Animated.Value(props.loaded ? 1 : 0),
+      loadingIndicatorOpacity: new Animated.Value(props.loaded ? 0 : 1),
+    };
+  }
+
+  componentWillReceiveProps = newProps => {
+    if (!this.props.loaded && newProps.loaded) {
+      Animated.timing(this.state.imageOpacity, {
+        toValue: 1,
+        duration: animationDurationBase,
+      }).start();
+      // once the image has finished fading in, we can stop
+      // rendering the spinner and background
+      setTimeout(() => {
+        Animated.timing(this.state.loadingIndicatorOpacity, {
+          toValue: 0,
+          duration: animationDurationBase,
+        }).start();
+      }, animationDurationBase);
+      setTimeout(() => {
+        this.setState({ showLoadingIndicator: false });
+      }, 2 * animationDurationBase);
+    }
+  };
+
+  onImageLoad = () => {
+    if (this.props.onLoad) {
+      this.props.onLoad();
+    }
+  };
+
+  render() {
+    const {
+      style: userStyle,
+      source,
+      alt,
+      inView,
+      borderRadius,
+      loaded,
+      ...rest
+    } = this.props;
+
+    const outerStyle = [styles.outer];
+    if (borderRadius) {
+      outerStyle.push(styles.outerWithBorderRadius);
+    }
+    if (userStyle) {
+      outerStyle.push(userStyle);
+    }
+
+    return (
+      <View style={outerStyle} {...rest}>
+        {this.state.showLoadingIndicator && (
+          <Animated.View
+            style={[
+              styles.loadingIndicatorView,
+              { opacity: this.state.loadingIndicatorOpacity },
+            ]}
+          >
+            {inView && <BpkSpinner small type="dark" />}
+          </Animated.View>
+        )}
+        {inView && (
+          <Animated.Image
+            onLoad={this.onImageLoad}
+            style={[styles.image, { opacity: this.state.imageOpacity }]}
+            source={source}
+            alt={alt}
```

Comment:
Because we're putting `...rest` on here now, you can remove source={source} and alt={alt} because `...rest` will encapsulate those.